### ISSUE TITLE
BlenderBIM - IFCPROPERTYSINGLEVALUE not inheriting "Description" from IFCSIMPLEPROPERTYTEMPLATE when exporting to ifc

### ISSUE THREAD

**Issue by vulevukusej** [IfcOpenShell/IfcOpenShell#1862]
When adding properties based on a template, the property added does not inherit "Description" after exporting to ifc.
Template:
![image](https://user-images.githubusercontent.com/83825269/140715639-c428dd86-c433-4954-9ee4-0e102114ad45.png)
Property Instance:
![image](https://user-images.githubusercontent.com/83825269/140715689-1eed688a-a024-4dcd-bf2e-8db274bc79e1.png)

**Comment by vulevukusej** [comment]
In this case, my expectation was that the property instance also has the Description "Tragendes Bauteil"

**Comment by Moult** [comment]
Yeah I'm not so sure about this - should it inherit the description, or is this description meant for a different human explanation? Ping @aothms

**Comment by aothms** [comment]
Don't ask me ;)

> The template may determine the name, description, data type, the unit, or a standard expression for each property that is based on that template.

https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/schema/ifckernel/lexical/ifcpropertytemplatedefinition.htm

I would guess Name and Description should be treated the same based on above snippet. But my guess is as good as yours.

**Comment by vulevukusej** [comment]
Do we expect use-cases where the template description for a property deviates from the property instance?  That seems counter-intuitive to me

**Comment by Moult** [comment]
I don't know. I intuitively expected description to be different - in the template the description describes the template name, and in the instance the description describes the property value. In theory (which I haven't implemented yet) the template is kept with the project and linked to the property, so copy pasting the description everywhere sounds like bad data practice.

**Comment by aothms** [comment]
> and in the instance the description describes the property value

The description for IfcProperty.Description is:

>  Informative text to explain the property. 

https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/schema/ifcpropertyresource/lexical/ifcproperty.htm

So I'm not so sure if it's about the value. Due to the order Name, Desc, Value, I always assumed Desc is about the Name.

A description for the value does sound useful though in theory...

**Comment by vulevukusej** [comment]
Ok, I agree with you regarding bad data practice, but I think a lookup of sorts is needed, whereby the returned value is the description from the template.

**Comment by Moult** [comment]
Hmm yeah given the current description for Description in the docs, I come to the same conclusion. I still think it's the wrong thing to do so I'm happy to disobey it and instead propose for the description to not have to copypaste the template.

For a lookup, that seems to be served by `IfcRelDefinesByTemplate` but I haven't implemented that.

**Comment by BIMvoice** [comment]
@vulevukusej This was resolved, though not the way the title suggests. `IfcRelDefinesByTemplate` links a whole Pset to a Pset Template, not individual properties to their template's description, so implementing it wouldn't have solved the actual ask here. Instead Bonsai does a live, name-matched lookup: a property's tooltip shows its template's Description (`bonsai/bim/helper.py::add_attribute_description`), falling back to the property template when it's not a standard buildingSMART Pset. That covers newly added properties since 2022, and Andrej730's `df4f565176` later closed the remaining gap for already-saved psets, which matches your exact repro.

Verified live: reopening an existing saved custom Pset with a described property now shows that description on hover, on current v0.8.0. Closing as resolved, feel free to reopen if this still doesn't match what you're seeing.
